File: backend/src/main/java/com/legaltrack/service/AdminService.java
package com.legaltrack.service;

import com.legaltrack.dto.admin.CaseSyncLogDto;
import com.legaltrack.dto.admin.SystemStatsDto;
import com.legaltrack.dto.auth.UserProfileDto;
import com.legaltrack.dto.common.PagedResponse;
import com.legaltrack.enums.RoleType;
import com.legaltrack.enums.UserStatus;
import org.springframework.data.domain.Pageable;

public interface AdminService {
    SystemStatsDto getSystemStats();
    PagedResponse<UserProfileDto> getUsers(RoleType role, Pageable pageable);
    void updateUserStatus(Long userId, UserStatus status);
    PagedResponse<CaseSyncLogDto> getSyncLogs(Pageable pageable);
}

File: frontend/src/services/adminService.js
import api from './api';

export const adminService = {
  getSystemStats: async () => {
    const res = await api.get('/admin/stats');
    return res.data;
  },

  getUsers: async (role, page = 0, size = 15) => {
    const res = await api.get(`/admin/users?${role ? `role=${role}&` : ''}page=${page}&size=${size}`);
    return res.data;
  },

  updateUserStatus: async (userId, status) => {
    const res = await api.patch(`/admin/users/${userId}/status?status=${status}`);
    return res.data;
  },

  getSyncLogs: async (page = 0, size = 20) => {
    const res = await api.get(`/admin/sync-logs?page=${page}&size=${size}`);
    return res.data;
  },

  getAuditLogs: async (page = 0, size = 20) => {
    const res = await api.get(`/audit-logs?page=${page}&size=${size}`);
    return res.data;
  }
};
